source/extensions/filters/udp/dns_filter/dns_parser.cc

```diff
@@ -553,11 +553,14 @@ void DnsMessageParser::buildResponseBuffer(DnsQueryContextPtr& query_context,
 
     const auto& answers = query_context->answers_;
 
-    // Generate a set of up to 8 random indices of the answer records that we serialize. We do this
-    // so that we don't always return the same records in the same order for every query.
-    absl::flat_hash_set<size_t> indices{};
-    const size_t max_indices = std::min(answers.size(), MAX_RETURNED_RECORDS);
-    generateRandomIndices(max_indices, indices);
+    // Generate a vector containing up to 8 indices of the answer records. We randomize the
+    // first entry if there are more than 8 records.
+    std::vector<size_t> indices(answers.size());
```

Comment:
Why do we need a new vector here? Can you just pick a random number and start from that index? Also, I think you only need to do this if the size is > max?

Reply:
I think I have a way to do this without needing a vector/set.